sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/kinesis/enhancedfanout/EFOShardSubscribersPool.java

```diff
@@ -235,12 +251,11 @@ KinesisRecord getNextRecord() throws IOException {
    * <p>In case of re-sharding, start all new {@link EFOShardSubscriber#subscribe subscriptions}
    * with the subscription {@link #errorHandler} if there is no {@link #subscriptionError} yet.
    */
-  @SuppressWarnings("FutureReturnValueIgnored")
   private void onEventDone(ShardState shardState, EventRecords noRecordsEvent) {
     if (noRecordsEvent.event.continuationSequenceNumber() == null
-        && noRecordsEvent.event.hasChildShards()
-        && subscriptionError == null) {
```

Comment:
It's important to also apply re-shard events if we're in the error state, otherwise we would lose this information in the checkpoints. The difference is that we don't start the subscription in this case.